lib/jobs/dell-wsman-configure-bios.js

```diff
@@ -0,0 +1,151 @@
+// Copyright 2017, DELL, Inc.
+
+'use strict';
+
+var di = require('di'),
+    urlParse = require('url-parse');
+
+module.exports = DellWsmanConfigureBiosFactory;
+di.annotate(DellWsmanConfigureBiosFactory, new di.Provide('Job.Dell.Wsman.ConfigureBios'));
+di.annotate(DellWsmanConfigureBiosFactory, new di.Inject(
+    'Job.Base',
+    'Logger',
+    'Promise',
+    'Assert',
+    'Util',
+    'Services.Waterline',
+    'Services.Configuration',
+    '_',
+    'HttpTool',
+    'Errors',
+    'Services.Encryption',
+    'validator'
+));
+
+function DellWsmanConfigureBiosFactory(
+    BaseJob,
+    Logger,
+    Promise,
+    assert,
+    util,
+    waterline,
+    configuration,
+    _,
+    HttpTool,
+    errors,
+    encryption,
+    validator
+) {
+    var logger = Logger.initialize(DellWsmanConfigureBiosFactory);
+
+    function DellWsmanConfigureBiosJob(options, context, taskId) {
+        DellWsmanConfigureBiosJob.super_.call(this, logger, options, context, taskId);
+        assert.object(this.options);
+        this.nodeId = this.context.target;
+    }
+
+    util.inherits(DellWsmanConfigureBiosJob, BaseJob);
+
+    DellWsmanConfigureBiosJob.prototype._run = function () {
+        var self = this;
+        self.dell = configuration.get('dell');
+
+        if (!self.dell || !self.dell.services || !self.dell.services.configuration) {
+            throw new errors.NotFoundError('Dell SCP  web service is not defined in smiConfig.json.');
+        }
+        return Promise.resolve(self.checkOBM())
+            .then(function(obm){
+                return self.configureBios(obm);
+            })
+            .then(function(){
+                return self._done();
+            })
+            .catch(function(err) {
+                self._done(err);
+            });
+    };
+
+    DellWsmanConfigureBiosJob.prototype.checkOBM = function() {
+        var self = this;
+        logger.info('checkOBM: Self.nodeID: ' + self.nodeId);
+        return waterline.nodes.findByIdentifier(self.nodeId)
+            .then(function(result) {
+                self.nodeType = result.type;
+                if (self.nodeType !== 'compute') {
+                    logger.info('SCP ConfigureBios is not applicable to node type ' + self.nodeType);
+                    self.cancel();
```

Comment:
`self.cancel` is designed to be called in baseTask or Task level, the job don't need to cancel itself. You just need to threw an error, the task engine will cancel the task.